### PR TITLE
fix: MITaskSchema and use vitest to run tests

### PR DESCRIPTION
**Pull request by ailinvenerus**
**Done:**
- Fix schema _MITaskSchema_: add _modelFilename_ field
- Update _package.json_ to run correctly with vitest

### CODE REVIEW

File: src/schema.ts [hunk 1]
```diff
@@ -44,9 +44,10 @@ const MITaskSchema = object({
   clientFilename: string().optional(),
   originalFilename: string().optional(),
   clientModelFilename: string().optional(),
+  modelFilename: string().optional(),
   status: zenum(['SUCCESS', 'DELETED', 'PENDING_UPLOAD', 'PROCESSING']),
   submitted: number().int().positive(),
-  modelUrl: string().url().optional(),
+  modelUrl: string().optional(),
 }).strict();
 export type MITask = zinfer<typeof MITaskSchema>;
```
Comment (EHadoux) [inline src/schema.ts:52]: Is all of this in sync with what's in the `types` library?

Comment (ailinvenerus) [reply src/schema.ts:52]: No, I'll update the `types` library since it's not going to work with the `.url()`.

Comment (EHadoux) [reply src/schema.ts:52]: The `types` library is the source of truth. You update it, you break the platform.

**Review by EHadoux — APPROVED** [review]

**Comment by alexbostock** [comment]
nitpick: have we considered instead (or also) removing `.strict()` from the schema?

Because `modelFilename` is not useful to clients (it is entirely derived from `jobid` and `client`, and only used when accessing S3 directly).

And making the schema less strict would allow us to add more fields to the API response later without breaking this library.

**Comment by EHadoux** [comment]
Superseded by https://github.com/ScribeLabsAI/ScribeMiNode/pull/6

**Pull request closed without merging**